packages/typescript/src/context/ts-context.ts

```diff
@@ -0,0 +1,16 @@
+import { ComponentContext, createContext, useContext } from "@alloy-js/core";
+
+export interface TypeScriptContext {
+  /** If in a type context */
+  type?: boolean;
+}
+
+/**
+ * Provides scopes for instance and static private members.
+ */
+export const TypeScriptContext: ComponentContext<TypeScriptContext> =
+  createContext<TypeScriptContext>();
+
+export function useTSContext(): TypeScriptContext {
+  return useContext(TypeScriptContext) ?? { type: false };
```

Comment:
You can set the default as the first argument to createContext.

Reply:
created a new component that hides that, let me know what you think